src/common/libsubprocess/subprocess.c

```diff
@@ -721,6 +725,127 @@ flux_subprocess_t *flux_rexec (flux_t *h, int rank, int flags,
     return NULL;
 }
 
+int flux_subprocess_stream_start (flux_subprocess_t *p, const char *stream)
+{
+    struct subprocess_channel *c;
+    flux_buffer_t *fb;
+
+    if (!p || p->magic != SUBPROCESS_MAGIC || (p->local && p->in_hook)) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (!stream)
```

Comment:
Seems like a NULL stream should be an EINVAL.  If used, it would only obfuscate the code at the call site.

Reply:
most of `libsubprocess` has this semantic.  I cannot remember how I arrived at this point, but perhaps it is no longer a good idea.  I'll leave this in here for this PR, but I'll create another issue to cover it for all of `libsubprocess`.

Reply:
is now issue #2288